src/snowflake/cli/plugins/nativeapp/codegen/snowpark/python_processor.py

```diff
@@ -179,8 +187,10 @@ def process(
             bundle_map, processor_mapping
         )
 
-        collected_output = []
+        dest_file_py_file_to_ddl_map: Dict[Path, str] = {}
```

Comment:
nit: I'm not sure we need to buffer. We can just write the SQL files for each python file for which generation succeeded. If we do this, we can populate the main file at the very end, we just need to remember the SQL files to `execute immediate` from the main file. That's a lot less information in memory for users that have thousands of UDFs/SPROCs.